src/Framework/ListTable/ModelColumn.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Give\Framework\ListTable;
+
+use Give\Framework\Support\Contracts\Arrayable;
+
+/**
+ * @template M of Give\Framework\Models\Model
+ */
+abstract class ModelColumn implements Arrayable
+{
+    /**
+     * @var string|array Define the meta key to be used when sorting the query by this column
+     */
+    protected $sortColumn;
+
+    /**
+     * @var bool Define if the column is visible
+     */
+    protected $visibleColumn;
+
+    /**
+     * Returns the id for that column.
+     *
+     * @unreleased
+     */
+    abstract public static function getId();
+
+    /**
+     * Returns the label for that column.
+     *
+     * @unreleased
+     */
+    abstract public function getLabel();
+
+    /**
+     * Returns the value to be displayed in the specific cell for that column and row.
+     *
+     * @unreleased
+     *
+     * @param M $model
+     */
+    abstract public function getCellValue($model);
+
+    /**
+     * @return bool
+     */
+    public function isSortable(): bool
+    {
+        return null !== $this->sortColumn;
+    }
+
+    /**
+     * @return array
+     */
+    public function getSortColumn(): array
+    {
+        if ( ! $this->isSortable() ) {
+            return [];
+        }
+
+        return is_array( $this->sortColumn ) ? $this->sortColumn : [$this->sortColumn];
+    }
+
+    /**
+     * @param bool $visible
+     *
+     * @return void
+     */
+    public function visible(bool $visible)
+    {
+        $this->visibleColumn = $visible;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isVisible(): bool
+    {
+        return $this->visibleColumn;
+    }
```

Comment:
These aren't needed anymore, right?

Reply:
I brought this into the ModelColumn so we can dynamically get/set column visibility based on the array of column ids.

Used on https://github.com/impress-org/givewp/pull/6413/files#diff-60d8eca599d35dccb6101103f4b43433c9dd60fe0a0c6d154f8cd043ea01d358R158

Reply:
Got it. This is fine, then.